argoverse/utils/make_att_files.py

```diff
@@ -0,0 +1,365 @@
+                    else:
+                        continue
+
+                    if dict_tracks[id_track]["ind_lidar_min"] == -1:
+                        dict_tracks[id_track]["ind_lidar_min"] = ind_lidar
+
+                    if (
+                        dict_tracks[id_track]["ind_lidar_max"] == -1
+                        or ind_lidar > dict_tracks[id_track]["ind_lidar_max"]
+                    ):
+                        dict_tracks[id_track]["ind_lidar_max"] = ind_lidar
+
+                    center = np.array([box["center"]["x"], box["center"]["y"], box["center"]["z"]])
+                    egovehicle_to_city_se3 = argoverse_loader.get_pose(ind_lidar, id_log)
+                    if egovehicle_to_city_se3 is None:
+                        print("Pose not found!")
+                        continue
+                    center_w = egovehicle_to_city_se3.transform_point_cloud(center[np.newaxis, :])[0]
+
+                    dict_tracks[id_track]["list_center"][ind_lidar] = center
+                    dict_tracks[id_track]["list_center_w"][ind_lidar] = center_w
+                    dict_tracks[id_track]["list_dist"][ind_lidar] = np.linalg.norm(center[0:2])
+                    dict_tracks[id_track]["exists"][ind_lidar] = True
+                    dict_tracks[id_track]["list_city_se3"][ind_lidar] = egovehicle_to_city_se3
+                    dict_tracks[id_track]["list_bbox"][ind_lidar] = box
+
+                length_track = dict_tracks[id_track]["ind_lidar_max"] - dict_tracks[id_track]["ind_lidar_min"] + 1
+                if dict_tracks[id_track]["ind_lidar_max"] == -1 and dict_tracks[id_track]["ind_lidar_min"] == -1:
+                    # this shouldn't happen
+                    dict_tracks[id_track]["length_track"] = 0
+                else:
+                    dict_tracks[id_track]["length_track"] = length_track
+
+                dict_tracks[id_track]["list_vel"], dict_tracks[id_track]["list_acc"] = compute_v_a(
+                    dict_tracks[id_track]["list_center_w"]
+                )
+                dict_tracks[id_track]["num_missing"] = (
+                    dict_tracks[id_track]["length_track"] - dict_tracks[id_track]["exists"].sum()
+                )
+                dict_tracks[id_track]["difficult_att"] = []
+                vel_abs = np.linalg.norm(dict_tracks[id_track]["list_vel"][:, 0:2], axis=1)
+                acc_abs = np.linalg.norm(dict_tracks[id_track]["list_acc"][:, 0:2], axis=1)
+
+                dist_close = 50
+                ind_valid = np.nonzero(1 - np.isnan(dict_tracks[id_track]["list_dist"]))[0]
+                ind_close = np.nonzero(dict_tracks[id_track]["list_dist"][ind_valid] < dist_close)[0]
+
+                if len(ind_close) > 0:
+                    ind_close_max = ind_close.max() + 1
+                    ind_close_min = ind_close.min()
+
+                if dict_tracks[id_track]["length_track"] == 0:
+                    # this shouldn't happen
```

Comment:
should we `raise Exception` here or `raise AssertionError` to catch this potential error?